src/pudl_archiver/archivers/epacems.py

```diff
@@ -86,5 +89,56 @@ async def get_year_quarter_resource(
             f"File no. {request_count}: Downloaded {year} Q{quarter} EPA CEMS hourly emissions data."
         )
         return ResourceInfo(
-            local_path=archive_path, partitions={"year_quarter": f"{year}q{quarter}"}
+            local_path=archive_path,
+            partitions={"year": year, "quarter": quarter},
+            layout=ZipLayout(filepaths=[filename]),
         )
+
+    def dataset_validate_archive(
+        self,
+        baseline_datapackage: DataPackage | None,
+        new_datapackage: DataPackage,
+        resources: dict[str, ResourceInfo],
+    ) -> list[validate.DatasetSpecificValidation]:
+        """Look for consecutive year quarter partitions in CEMS archive."""
+        year_quarters = [x.parts["year_quarter"] for x in new_datapackage.resources]
+        yq_split = [x.split("q") for x in year_quarters]
+        yq_df = pd.DataFrame(yq_split, columns=["year", "quarter"]).apply(
+            pd.to_numeric, errors="coerce"
+        )
+        newest_year = yq_df.year.max()
```

Comment:
The idea here is to allow for incomplete (but non non-consecutive) data in the newest year. This is because we want to be able to ingest the data one quarter at a time, as it comes out. For all years prior to the newest year, we expect all four quarters to be available. 

This would only cause an issue if we pulled a subset of the data and there happened to be quarters missing from the last pulled year when there should be complete data. This function would not flag those missing quarters because it enables incomplete "last" years.

Given that we haven't really used the `only_years` flag, this doesn't seem like a huge issue. However, I see how it could potentially cause trouble. I agree that puling the current date and performing some math to get the appropriate lag time is probably more likely to cause problems than leaving it as is.